tiatoolbox/cli/semantic_segment.py

```diff
@@ -0,0 +1,174 @@
+"""Command line interface for semantic segmentation."""
+import os
+import pathlib
+
+import click
+import yaml
+
+from tiatoolbox import utils
+from tiatoolbox.models.controller.semantic_segmentor import (
+    IOSegmentorConfig,
+    SemanticSegmentor,
+)
+
+
+@click.group()
+def main():  # pragma: no cover
+    """Define slide_info click group."""
+    return 0
+
+
+@main.command()
+@click.option(
+    "--pretrained_model",
+    help="Predefined model used to process the data. the format is "
+    "<model_name>_<dataset_trained_on>. For example, `resnet18-kather100K` is a "
+    "resnet18 model trained on the kather dataset. For a detailed list of "
+    "available pretrained models please see "
+    "https://tia-toolbox.readthedocs.io/en/latest/usage.html"
+    "#tiatoolbox.models.classification.patch_predictor.get_pretrained_model",
+    default="fcn-tissue_mask",
+)
+@click.option(
+    "--pretrained_weights",
+    help="Path to the model weight file. If not supplied, the default "
+    "pretrained weight will be used.",
+    default=None,
+)
+@click.option(
+    "--img_input",
+    help="Path to the input directory containing images to process or an "
+    "individual file.",
+)
+@click.option(
+    "--file_types",
+    help="File types to capture from directory. "
+    "default='*.png', '*.jpg', '*.jpeg', '*.tif', '*.tiff'",
+    default="*.png, *.jpg, *.jpeg, *.tif, *.tiff, *.svs, *.ndpi, *.jp2, *.mrxs",
+)
+@click.option(
+    "--masks",
+    help="Path to the input directory containing masks to process corresponding to "
+    "image tiles and whole-slide images. Patches are only processed if they are "
+    "within a masked area. If masks are not provided, then a tissue mask will be "
+    "automatically generated for whole-slide images or the entire image is "
+    "processed for image tiles. Supported file types are jpg, png and npy.",
+    default=None,
+)
```

Comment:
Not .npy at the moment. I think you forgot to change it after our discussion in the chat.